test/plugins/TPSetBufferCreator.cpp

```diff
@@ -0,0 +1,336 @@
+  m_thread.start_working_thread("buffer-man");
+  TLOG() << get_name() << " successfully started";
+  TLOG() << get_name() << ": Exiting do_start() method";
+}
+
+void
+TPSetBufferCreator::do_stop(const nlohmann::json& /*args*/)
+{
+  TLOG() << get_name() << ": Entering do_stop() method";
+  m_thread.stop_working_thread();
+  TLOG() << get_name() << " successfully stopped";
+
+  size_t sentCount = 0;
+  TPSetBuffer::data_request_output requested_tpset;
+  if (m_dr_on_hold.size()) { // check if there are still data request on hold
+    TLOG() << get_name() << ": On hold DRs: " << m_dr_on_hold.size();
+    std::map<dfmessages::DataRequest, std::vector<trigger::TPSet>>::iterator it = m_dr_on_hold.begin();
+    while (it != m_dr_on_hold.end()) {
+
+      requested_tpset.txsets_in_window = it->second;
+      std::unique_ptr<dataformats::Fragment> frag_out = convert_to_fragment(requested_tpset, it->first);
+      TLOG() << get_name() << ": Sending late requested data (" << (it->first).window_begin << ", "
+             << (it->first).window_end << "), containing " << requested_tpset.txsets_in_window.size() << " TPSets.";
+
+      if (requested_tpset.txsets_in_window.size()) {
+        frag_out->set_error_bit(dataformats::FragmentErrorBits::kIncomplete, true);
+      } else {
+        frag_out->set_error_bit(dataformats::FragmentErrorBits::kDataNotFound, true);
+      }
+
+      send_out_fragment(std::move(frag_out));
+      sentCount++;
+      it++;
+    }
+  }
+
+  m_tps_buffer->clear_buffer(); // emptying buffer
+
+  TLOG() << get_name() << ": Exiting do_stop() method : sent " << sentCount << " incomplete fragments";
+}
+
+void
+TPSetBufferCreator::do_scrap(const nlohmann::json& /*args*/)
+{
+  TLOG() << get_name() << ": Entering do_scrap() method";
+  m_tps_buffer.reset(nullptr); // calls dtor
+  TLOG() << get_name() << ": Exiting do_scrap() method";
+}
+
+std::unique_ptr<dataformats::Fragment>
+TPSetBufferCreator::convert_to_fragment(TPSetBuffer::data_request_output /* ds_output */,
+                                        dfmessages::DataRequest input_data_request)
+{
+  auto ret = std::make_unique<dataformats::Fragment>(std::vector<std::pair<void*, size_t>>());
+  auto& frag = *ret.get();
+
+  dataformats::GeoID geoid(dataformats::GeoID::SystemType::kDataSelection, m_conf.region, m_conf.element);
+  dataformats::FragmentHeader frag_h;
+  frag_h.trigger_number = input_data_request.trigger_number;
+  frag_h.trigger_timestamp = input_data_request.trigger_timestamp;
+  frag_h.window_begin = input_data_request.window_begin;
+  frag_h.window_end = input_data_request.window_end;
+  frag_h.run_number = input_data_request.run_number;
+  frag_h.element_id = geoid;
+  frag_h.fragment_type = (dataformats::fragment_type_t)dataformats::FragmentType::kTriggerPrimitives;
+  frag_h.sequence_number = input_data_request.sequence_number;
+
+  frag.set_header_fields(frag_h);
+
+  /*  >>> need to fill the fragment with
```

Comment:
Is this bit post 2.8?

Reply:
Yes. It is not settled yet how the fragments should be filled. Returning empty ones for now.